internal.sh

```diff
@@ -0,0 +1,86 @@
+rm -fr crc-cluster-kube-controller-manager-operator
+
+function run_preflight_checks() {
+        if [ -z "${OPENSHIFT_PULL_SECRET_PATH-}" ]; then
+            echo "OpenShift pull secret file path must be specified through the OPENSHIFT_PULL_SECRET_PATH environment variable"
+            exit 1
+        elif [ ! -f ${OPENSHIFT_PULL_SECRET_PATH} ]; then
+            echo "Provided OPENSHIFT_PULL_SECRET_PATH (${OPENSHIFT_PULL_SECRET_PATH}) does not exists"
+            exit 1
+        fi
+}
+
+run_preflight_checks
+
+MIRROR=${MIRROR:-https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp}
+
+# If user defined the OPENSHIFT_VERSION environment variable then use it.
+if test -n "${OPENSHIFT_VERSION-}"; then
+    OPENSHIFT_RELEASE_VERSION=${OPENSHIFT_VERSION}
+    echo "Using release ${OPENSHIFT_RELEASE_VERSION} from OPENSHIFT_VERSION"
+else
+    OPENSHIFT_RELEASE_VERSION="$(curl -L "${MIRROR}"/candidate-4.11/release.txt | sed -n 's/^ *Version: *//p')"
+    if test -n "${OPENSHIFT_RELEASE_VERSION}"; then
+        echo "Using release ${OPENSHIFT_RELEASE_VERSION} from the latest mirror"
+    else
+        echo "Unable to determine an OpenShift release version.  You may want to set the OPENSHIFT_VERSION environment variable explicitly."
+        exit 1
+    fi
+fi
+
+if test -z "${OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE-}"; then
+    OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE="$(curl -L "${MIRROR}/${OPENSHIFT_RELEASE_VERSION}/release.txt" | sed -n 's/^Pull From: //p')"
+elif test -n "${OPENSHIFT_VERSION-}"; then
+    echo "Both OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE and OPENSHIFT_VERSION are set, OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE will take precedence"
+    echo "OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE: $OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE"
+    echo "OPENSHIFT_VERSION: $OPENSHIFT_VERSION"
+fi
+echo "Setting OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE to ${OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE}"
+
+mkdir -p openshift-clients/linux
+curl -L "${MIRROR}/${OPENSHIFT_RELEASE_VERSION}/openshift-client-linux-${OPENSHIFT_RELEASE_VERSION}.tar.gz" | tar -zx -C openshift-clients/linux oc
+OC=./openshift-clients/linux/oc
+
+function patch_and_push_image() {
+    local image_name=$1
+    image=$(${OC} adm release info -a ${OPENSHIFT_PULL_SECRET_PATH} ${OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE} --image-for=${image_name})
+    vcs_ref=$(${OC} image info -a ${OPENSHIFT_PULL_SECRET_PATH} ${image} -ojson | jq -r '.config.config.Labels."vcs-ref"')
+    url=$(${OC} image info -a ${OPENSHIFT_PULL_SECRET_PATH} ${image} -ojson | jq -r '.config.config.Labels.url')
+    url_tag=$(basename ${url})
+    rhpkg clone containers/crc-${image_name}
+    pushd crc-${image_name}
+    git remote add upstream git://pkgs.devel.redhat.com/containers/ose-${image_name}
+    git fetch --all
+    git checkout --track  origin/rhaos-4.11-rhel-8
+    git merge ${vcs_ref}
+    git push -f origin HEAD
+    rhpkg container-build  --target crc-1-rhel-8-candidate
```

Comment:
My feeling is that we need targets/branches which follow openshift versioning, `crc-4.11-rhel-8-candidate` and such

Reply:
@cfergeau do we need to request for that as of now we only have `crc-1-rhel-8-candidate` tag to perform tag build.

Reply:
This shouldn't blocker for this PR and need bit more discussion, if we are going with `crc-4.11-rhel-8-candidate` which means every openshift release version we need to request a new target.

Reply:
But do you think this makes sense, and that it is something we should keep exploring? Or is it a bad idea/not useful because ..., and we can stop the discussion now?